### PR TITLE
Add Kubernetes autoscaling, liveness probe, and dry run

### PR DESCRIPTION
**Pull request by adammcmaster**
This sets up autoscaling with what I think are sensible limits. It also configures a liveness probe to ensure deploys don't cause any downtime (old containers won't be taken down until the new ones are live).

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: kubernetes/deployment-production.tmpl [hunk 1]
```diff
@@ -38,6 +37,10 @@ spec:
             limits:
               memory: "2200Mi"
               cpu: "500m"
+          livenessProbe:
+            httpGet:
+              path: /
```
Comment (camallen) [inline kubernetes/deployment-production.tmpl:42]: i'm not 100% sure the app returns a 200 response on this path...worth asking dev team for this.

File: kubernetes/deployment-production.tmpl [hunk 2]
```diff
@@ -96,6 +111,10 @@ spec:
             limits:
               memory: "275Mi"
               cpu: "500m"
+          livenessProbe:
+            httpGet:
+              path: /
```
Comment (camallen) [inline kubernetes/deployment-production.tmpl:116]: Again we need to make sure this path responds with 200 else it'll kill the pod.

**Comment by adammcmaster** [comment]
👍 I've added readiness probes now and will wait for word on how the app responds to `GET /`

**Comment by adammcmaster** [comment]
I've just done a quick test and confirmed that `GET /` won't work. Options are:

1. Hardcode some paths that do work – might break if those pages move in the future.
2. Add a dedicated `/_liveness` (or similar) route to each app – probably the best option long term.
3. Change the probes to use `tcpProbe` to just check the app is listening, without actually making a request – doesn't check the app is actually healthy, but better than nothing.

I'm tempted to go with #3 for now just for speed, so that we can get autoscaling deployed this week (I won't have time to implement #2 before the next deploy).

**Comment by adammcmaster** [comment]
Yup that's right, it'll run those every time. The idea is to make the PR status check fail if there's a syntax error in the deployment template. (I added this in a few other repos last week after I opened/merged a PR with a syntax error that failed to deploy.)

**Comment by adammcmaster** [comment]
I've updated the probes to use `tcpSocket` instead of `httpGet`. We can change that later if we want to.

**Comment by eatyourgreens** [comment]
I haven't checked the content pages app, but there's an index page set up for the projects server.
https://github.com/zooniverse/front-end-monorepo/blob/master/packages/app-project/pages/Index.js

In theory, you should get that back from a request for `/` for the projects app.

**Comment by eatyourgreens** [comment]
It turns out macOS is case-insensitive, so `/Index` is served as `/` for local development on a mac.

Both apps have a page at `/Index` which can be used for the health check.

File: kubernetes/deployment-production.tmpl [hunk 3]
```diff
@@ -96,6 +113,12 @@ spec:
             limits:
               memory: "275Mi"
               cpu: "500m"
+          livenessProbe:
+            tcpSocket:
+              port: 3000
```
Comment (camallen) [inline kubernetes/deployment-production.tmpl:118]: I assume we're switching to use the `/Index` path via HTTP instead of tcp?

Comment (adammcmaster) [reply kubernetes/deployment-production.tmpl:118]: Yup, going to switch that over this afternoon.

**Pull request merged**